arbeitszeit/records.py

```diff
@@ -225,7 +226,11 @@ def expected_sales_value(self) -> Decimal:
 
     @property
     def is_approved(self) -> bool:
-        return self.approval_date is not None
+        return self.approval_date is not None and self.rejection_date is None
```

Comment:
thanks @Amittai-Aviram, I think this is a valid point. 

> Can a plan be both accepted and rejected?
currently, then answer is "no". However, I think I chose to use this new column "rejection_date" under the impression of our first discussion in which the tendency seemed to be that maybe in the future this could change, i.e. one accountant accepts a plan while another one rejects it, and then maybe at this point there will be a third column "final_decision_date". In contrast to that, if we want to decide now that once a plan is rejected/approved, this cannot be changed anymore, then your suggestion might be a good solution. On the other hand, this would mean to basically re-start this PR since we would need a different db migration, i.e. changing column "approval_date" to "decision_date" and add new column "accepted". 
Also note that in the records.py file from above, the check I added is just for the sake of completeness, it is logically not really required since currently, once a plan is approved/rejected, in the UI it disappears from the "to review-list" so there is no chance that 
self.approval_date is not None and self.rejection_date is not None 
Overall, I think the approach with an extra column "rejection_date" gives a little more flexibility for the future, while in the records Plan class it does not make a big difference, all three states can be reflected with both approaches. 
But maybe we should discuss again how to go forward with PlanReview table.

Reply:
Thanks for the explanation.  Personally, I am uncomfortable with a DB schema that looks redundant or ambiguous, but maybe the team as a whole would agree with you to keep it this way.

Reply:
I agree that it would be better to have the three states Amittai is proposing. If changing an approved plan will be possible in the future and how this will be modelled is still very unclear. On the other hand this proposel would imply, on top of the changes to the current PR, changes in both DatabaseGateway implementation classes in `arbeitszeit_flask/database/repositories.py` and `tests/use_cases/repositories.py`. Some methods that would need to be changed (at first glance):

- `PlanQueryResult.ordered_by_activation_date`
- `PlanQueryResult.that_are_approved`
- `PlanQueryResult.that_were_activated_before`
- `PlanQueryResult.that_will_expire_after`
- `PlanQueryResult.that_are_expired_as_of`
- `PlanQueryResult.without_completed_review`
- `PlanUpdate.set_approval_date`

Moreover, a database migration would need to handle the change to the row `approval_date` in the database table model `PlanReview` in `arbeitszeit_flask/database/models.py`. 

Probably it's all managable, and perhaps it's not even very complicated,  but it's a lot for the current PR, isn't it? We could talk about it Tuesday or find a date for a dedicated call.

Reply:
picking up the threads again: I am fine to discuss this again and promise to put it on the agenda for next meeting :) Just to make my point from above clear again: the three-fold logic 

>  Accept: accepted == True and decision_date is not None
    Rejected: accepted == False and decsion_date is not None
    Pending: decision_date is None.

is technically equally reflected in the current implementation:
```
Accept: approval_date is not None and rejection_date is None
Rejected: approval_date is None and rejection_date is not None
Pending: approval_date is None and rejection_date is None
```

There is no ambiguity here. I think this is more a question of semantics. But this question of semantics also arises looking at the first approach because what will be the default value for "accepted" bool? Is a plan in "accepted == False" state at the beginning? If yes, it has a direct dependency to decision_date, i.e. order matters, and we will always have to evaluate decision_date first.